### PR TITLE
Add profile type to the DB

### PR DESCRIPTION
**Pull request by chmouel**
Add profile type into the DB using an ENUM types of 'free' and 'paid' (default:
'free'), should be easy to alter/modify if needed in the future.

Expose it to the /api/tenant call

### CODE REVIEW

**Comment by fabric8cd** [comment]
@chmouel snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-431-1`

**Comment by aslakknutsen** [comment]
[test]

**Comment by fabric8cd** [comment]
@chmouel snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-431-2`

File: migration/sql-files/003-profiles.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CREATE TYPE profiles AS ENUM ('free', 'paid');
```
Comment (aslakknutsen) [inline migration/sql-files/003-profiles.sql]: You need to register the migration here https://github.com/fabric8-services/fabric8-tenant/blob/master/migration/migration.go#L105

**Comment by fabric8cd** [comment]
@chmouel snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-431-4`

**Comment by aslakknutsen** [comment]
[test]

**Pull request merged**